smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/auth/AuthUtils.java

```diff
@@ -75,15 +76,16 @@ private AuthUtils() {}
     public static Map<ShapeId, HttpAuthScheme> getAllEffectiveNoAuthAwareAuthSchemes(
         ServiceShape serviceShape,
         ServiceIndex serviceIndex,
-        SupportedHttpAuthSchemesIndex authIndex
+        SupportedHttpAuthSchemesIndex authIndex,
+        TopDownIndex topDownIndex
     ) {
         Map<ShapeId, HttpAuthScheme> effectiveAuthSchemes = new TreeMap<>();
         var serviceEffectiveAuthSchemes =
             serviceIndex.getEffectiveAuthSchemes(serviceShape, AuthSchemeMode.NO_AUTH_AWARE);
         for (ShapeId shapeId : serviceEffectiveAuthSchemes.keySet()) {
             effectiveAuthSchemes.put(shapeId, authIndex.getHttpAuthScheme(shapeId));
         }
-        for (var operation : serviceShape.getAllOperations()) {
+        for (var operation : topDownIndex.getContainedOperations(serviceShape)) {
```

Comment:
is there anything going in place to warn that `serviceShape.getAllOperations()` is not how to get service operations? 😕

Reply:
Technically the [javadoc for ServiceShape::getAllOperations()](https://javadoc.io/doc/software.amazon.smithy/smithy-model/latest/software/amazon/smithy/model/shapes/EntityShape.html#getAllOperations()) mentions that it doesn't contain everything and references [`TopDownIndex::getContainedOperations()`](https://javadoc.io/doc/software.amazon.smithy/smithy-model/latest/software/amazon/smithy/model/knowledge/TopDownIndex.html#getContainedOperations(software.amazon.smithy.model.shapes.ToShapeId)), unfortunately there isn't a way to enforce it.

Reply:
Albeit there is [documentation on smithy.io](https://smithy.io/2.0/guides/building-codegen/using-the-semantic-model.html#get-every-operation-in-a-service-or-resource) that documents using the `TopDownIndex`.